instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -84,10 +89,16 @@ def client_attributes(sql = nil)
                 attributes[::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT] = sql
               end
             end
-
+            attributes.compact!
```

Comment:
I noticed other instrumentations were removing `nil` and thought Trilogy could benefit from this as well.

Reply:
Just micro-optimization on `compact!` like this [comment](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/693#discussion_r1360669417) and this [comment](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/887#discussion_r1586410848) that you can consider it or leave it as it is.

Reply:
Thanks @xuan-cao-swi! It looks like we are doing `nil` checks in places, so using `compact!` maybe wasn't that helpful to begin with! I've removed this and added a `nil` check for `db.collection.name` before adding the attribute. https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1109/commits/f9c244eaf83ee3e891748ad385363bc41644a48f